Assets/Scripts/Utils/TypedDictionary.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using System.Collections.Generic;
+using AreYouFruits.Nullability;
+
+namespace Growing.Utils
+{
+    public sealed class TypedDictionary<TBase>
```

Comment:
Why use separate TypedDictionary<TBase> instead of Dictionary<Type, TBase>? DRY?

Reply:
Thus there is no need to convert resolved object to the requested type. It is always of that type. And all this object->T and T->object logic is done inside TypedDictionary and is safe to use outside the TypedDictionary without any InvalidCastExcetions